helpers/integration.go

```diff
@@ -187,14 +187,13 @@ func CalculateIntegrationPipelineRunOutcome(logger logr.Logger, pipelineRun *tek
 	return true, nil
 }
 
-// GetLatestPipelineRunForSnapshotAndScenario returns the latest Integration PipelineRun for the
+// GetAllPipelineRunsForSnapshotAndScenario returns all Integration PipelineRun for the
 // associated Snapshot and IntegrationTestScenario. In the case the List operation fails,
 // an error will be returned.
-func GetLatestPipelineRunForSnapshotAndScenario(adapterClient client.Client, ctx context.Context, application *applicationapiv1alpha1.Application, snapshot *applicationapiv1alpha1.Snapshot, integrationTestScenario *v1alpha1.IntegrationTestScenario) (*tektonv1beta1.PipelineRun, error) {
+func GetAllPipelineRunsForSnapshotAndScenario(adapterClient client.Client, ctx context.Context, snapshot *applicationapiv1alpha1.Snapshot, integrationTestScenario *v1alpha1.IntegrationTestScenario) (*[]tektonv1beta1.PipelineRun, error) {
 	integrationPipelineRuns := &tektonv1beta1.PipelineRunList{}
-	var latestIntegrationPipelineRun = &tektonv1beta1.PipelineRun{}
 	opts := []client.ListOption{
-		client.InNamespace(application.Namespace),
```

Comment:
This change `s/application/snapshot` should be IMO explained in commit message

Reply:
Good catch, amended the commit message to mention that.